packages/core/src/wallet/Wallet.ts

```diff
@@ -24,7 +24,7 @@ export interface Wallet {
   initPublicDid(didConfig: DidConfig): Promise<void>
   createDid(didConfig?: DidConfig): Promise<DidInfo>
   pack(payload: Record<string, unknown>, recipientKeys: string[], senderVerkey?: string): Promise<EncryptedMessage>
-  unpack(encryptedMessage: EncryptedMessage): Promise<DecryptedMessageContext>
```

Comment:
We still use the term `EncryptedMessage` so if we want to rename this to `Unpacked` we should probably also call it `PackedMessage`. Any reason you want to go for pack vs encrypt? I'd be leaning to encrypted as it's a more commonly used term over packed, but fine to go with either of them, as long as we're consistent

Reply:
Ah, you got me. I just needed to do a small naming workaround to distinguish between the following data structures:

```
export interface DecryptedMessageContext {
  plaintextMessage: PlaintextMessage
  senderKey?: Key
  recipientKey?: Key
}
```

and 

```
export interface UnpackedMessageContext {
  plaintextMessage: PlaintextMessage
  senderKey?: string
  recipientKey?: string
}
```

I hesitated a bit where to draw the line between Indy types and AFJ types. Now, I see the `Wallet` interface as part of the AFJ so its types should be also part of the AFJ. Implementors, like `IndyWallet` can use different types inside. So we can use `DecryptedMessageContext` in the `Wallet` interface.

Regarding the `unpack/pack` vs `decrypt/encrypt`. I also tended towards using decrypt/encrypt, it seems more natural to me. But if I remember correctly, there were discussions about this in Indy/Aries community before and it has been changed from encrypt to pack. I don't know the exact reason. 

One thing where I see a small benefit of pack/unpack is that the message can be encrypted many times with the use of a forward message.

Reply:
There can be also a problem with dependencies. If we use `DecryptedMessageContext` we should move conversion from `publicKeyBase58` to instance of `Key` inside `IndyWallet` and that means a dependency on `modules/dids`

Reply:
I gave it another look and maybe we can live with the dependency for now.

Reply:
Looking at DIDComm v2 they're using encrypted:

Authcrypted and/or anoncrypted | application/didcomm-encrypted+json
-- | --
Signed and then anoncrypted | application/didcomm-encrypted+json
Signed | application/didcomm-signed+json
Plaintext | application/didcomm-plain+json


--- 

Re the dependency on `modules/dids` the `Key` class has been moved to the crypto directory in a separate PR (as it's not only related to dids). Would that resolve the issue?